src/NosCore.Controllers/Miniland/MinilandObjects/MgPacketHandler.cs

```diff
@@ -43,137 +43,136 @@ public class MgPacketHandler : PacketHandler<MinigamePacket>, IWorldPacketHandle
+                    await OpenGiftBatch();
                     break;
 
                 case 9:
-                    UseCoupon();
+                    await UseCoupon();
                     break;
             }
-            return Task.CompletedTask;
         }
 
-        private void UseCoupon()
+        private async Task UseCoupon()
         {
-            var item = _clientSession.Character.InventoryService.Select(s => s.Value)
-                .Where(s => (s.ItemInstance.ItemVNum == 1269) || (s.ItemInstance.ItemVNum == 1271)).OrderBy(s => s.Slot)
+            var item = _clientSession!.Character.InventoryService.Select(s => s.Value)
+                .Where(s => (s.ItemInstance?.ItemVNum == 1269) || (s.ItemInstance?.ItemVNum == 1271)).OrderBy(s => s.Slot)
                 .FirstOrDefault();
             if (item != null)
             {
-                var point = item.ItemInstance.ItemVNum == 1269 ? 300 : 500;
+                var point = item.ItemInstance!.ItemVNum == 1269 ? 300 : 500;
                 _clientSession.Character.InventoryService.RemoveItemAmountFromInventory(1, item.ItemInstance.Id);
-                _minilandObject.InventoryItemInstance.ItemInstance.DurabilityPoint += point;
-                _clientSession.SendPacket(new InfoPacket
+                _minilandObject!.InventoryItemInstance!.ItemInstance!.DurabilityPoint += point;
+                await _clientSession.SendPacket(new InfoPacket
                 {
                     Message = string.Format(Language.Instance.GetMessageFromKey(LanguageKey.REFILL_MINIGAME,
                         _clientSession.Account.Language), point)
                 });
-                ShowMinilandManagment();
+                await ShowMinilandManagment();
             }
         }
 
-        private void ShowMinilandManagment()
+        private Task ShowMinilandManagment()
         {
-            _clientSession.SendPacket(new MloMgPacket
+            return _clientSession!.SendPacket(new MloMgPacket
             {
-                MinigameVNum = _minigamePacket.MinigameVNum,
-                MinilandPoint = _miniland.MinilandPoint,
+                MinigameVNum = _minigamePacket!.MinigameVNum,
+                MinilandPoint = _miniland!.MinilandPoint,
                 Unknown1 = 0,
                 Unknown2 = 0,
-                DurabilityPoint = _minilandObject.InventoryItemInstance.ItemInstance.DurabilityPoint,
-                MinilandObjectPoint = _minilandObject.InventoryItemInstance.ItemInstance.Item.MinilandObjectPoint
+                DurabilityPoint = _minilandObject!.InventoryItemInstance!.ItemInstance!.DurabilityPoint,
+                MinilandObjectPoint = _minilandObject.InventoryItemInstance.ItemInstance.Item!.MinilandObjectPoint
             });
         }
 
-        private void OpenGiftBatch()
+        private async Task OpenGiftBatch()
         {
             var amount = 0;
-            switch (_minigamePacket.Point)
+            switch (_minigamePacket!.Point)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)